Source/palette.cpp

```diff
@@ -120,20 +120,22 @@ static void GenerateBlendedLookupTable(SDL_Color *palette, int skipFrom, int ski
 
 void LoadPalette(const char *pszFileName)
 {
-	int i;
-	void *pBuf;
-	BYTE PalData[256][3];
-
 	assert(pszFileName);
 
-	SFileOpenFile(pszFileName, &pBuf);
-	SFileReadFileThreadSafe(pBuf, (char *)PalData, sizeof(PalData));
-	SFileCloseFile(pBuf);
+	struct Color {
+		uint8_t r;
+		uint8_t g;
+		uint8_t b;
+	};

```

Comment:
You could add this to engine.h alongside Point.

Reply:
Shouldn't the fields be UpperCase too as per our convention?

I'd also suggest renaming to `Red`, `Green` and `Blue`.

Reply:
> Shouldn't the fields be UpperCase too as per our convention?

Where do you see that? Why?

By the way, we are [supposed](https://github.com/diasurgical/devilutionX/wiki/Code-Style) to follow the [Webkit](https://webkit.org/code-style-guidelines/) one, but we don't: class variables (not fields by the way, that's C#) should be prefixed by `m_`, no capitals on functions, etc.

Reply:
Maybe that Wiki would need an update @AJenbo? Or we should just follow that convention? The Webkit one is quite classic for C++. ^

Reply:
this is what we have atm:

https://github.com/diasurgical/devilutionX/blob/master/Source/.clang-tidy#L62

took a breake from applying it as there have been a lot of prs building up and people gt grumpy from having to rebase

Reply:
When you've had to type m_Green for the 50000th time, you really wish you'd just named the fields "r", "g" and "b".

Reply:
> When you've had to type m_Green for the 50000th time, you really wish you'd just named the fields "r", "g" and "b".

The `m_` prefix is only for private class variables. For cases like these I think these should be public and without getter/setter, and lowercase.